File: 04-estructuras-datos/08-arboles-binarios/java/ejemplos/ArbolBinarioBasico.java
public class ArbolBinarioBasico {
    static class Nodo {
        String valor;
        Nodo izquierdo;
        Nodo derecho;

        Nodo(String valor) { this.valor = valor; }
    }

    public static void main(String[] args) {
        Nodo raiz = new Nodo("Programación");

        // Cada referencia conecta un nodo con uno de sus hijos.
        raiz.izquierdo = new Nodo("Backend");
        raiz.derecho = new Nodo("Frontend");

        System.out.println("Raíz: " + raiz.valor);
        System.out.println("Hijo izquierdo: " + raiz.izquierdo.valor);
        System.out.println("Hijo derecho: " + raiz.derecho.valor);
    }
}

File: 04-estructuras-datos/08-arboles-binarios/scala/ejemplos/ArbolBinarioBasico.scala
case class Nodo(valor: String, izquierdo: Option[Nodo] = None, derecho: Option[Nodo] = None)

object ArbolBinarioBasico:
  def main(args: Array[String]): Unit =
    val raiz = Nodo("Programación", Some(Nodo("Backend")), Some(Nodo("Frontend")))
    // Option representa de forma explícita si existe o no un hijo.
    println(raiz.izquierdo.map(_.valor))
